### ISSUE TITLE
100% CPU in mockrunner JMS module

### ISSUE THREAD

**Issue by fabpicca** [mockrunner/mockrunner#13]
Hello,
I've noticed a strange behavior while developing a testing module for my application.
I'm using Spring JMS support.
When I start the application I have the CPU at 100%, and looks like I have an infinite loop on Session.commit() (by raising the debug level to DEBUG I can see the trace left here

```
public void commit() throws JMSException
{
     connection.throwJMSException();
     numberCommits++;
     if(logger.isDebugEnabled())
         logger.debug("Mock session commit");
}
```

Am I doing something wrong here?

**Comment by cemartins** [comment]
Hello,

There is no loop in that method so it cannot be the cause of your infinite
loop.
Maybe there is a loop down the stack trace that calls `session.commit()`
and for some reason never stops. You must find that piece of code.

kind regards
Carlos

**Comment by fabpicca** [comment]
Thanks Carlos for your prompt answer. 
Given that I cannot spot any "nasty" loops, looks like there are some bad interactions at transaction level. Do you have any visibility of similar issues?
Thanks!

**Comment by cemartins** [comment]
I suppose the DefaultMessageListenerContainer polls the queue which responds immediately and thus the loop.
Could you post a maven test case reproducing this problem? Or would you rather stick with your workaround and we close this issue?

**Comment by keilw** [comment]
Seems I was able to replicate that (in 1.0.8, v 1.0.4 would not return any message under the same circumstances) when trying to use a **setJMSCorrelationID**. Will file a separate issue for that particular problem.

**Comment by justinpitts** [comment]
I've tried to replicate this issue.

I have a simple test which manually ( not via DI ) configures a DMLC, starts it, and sends it 1E6 messages. At no time does my CPU exceed 14%. I have tried this both with and without a jms correlation id.

One curious thing that I notice is that the time to SEND messages increases with each message sent, at a rate of roughly 80ms/10k messages.

Does anyone have a working reproduction of this issue?
